go/v1beta1/storage/esutil/client.go

```diff
@@ -137,9 +146,32 @@ func (c *client) Create(ctx context.Context, request *CreateRequest) (string, er
 		indexOpts = append(indexOpts, c.esClient.Index.WithDocumentID(request.DocumentId))
 	}
 
+	var (
+		doc []byte
+		err error
+	)
+	if request.Join != nil {
```

Comment:
There's a similar block of serialization logic in `Bulk`, might be a good spot to extract a function.

Reply:
I tried this at first, but the logic between the two differs just enough to make the function not worth it, imo.  In particular, the way that routing is handled is different between the two.